src/os_handlers.eliom

```diff
@@ -207,9 +207,13 @@ let activation_handler_common ~akey =
      TODO: do not disconnect users if we relog them with the same userid.
   *)
   try%lwt
-    let%lwt {userid; email; validity; action = _; data = _; autoconnect} =
+    let%lwt {userid; email; validity; action; data = _; autoconnect} =
       Os_user.get_activationkey_info akey in
     let%lwt () =
+      if action = "activation" && validity <= 0L then
```

Comment:
I agree and thought about that before, and then gave up... It seemed to be not trivial, because I don't want OS to list all possible actions: I want the users of OS to be able to define and implement their own actions.

I'll think again about how to do this.